ibis/expr/api.py

```diff
@@ -3274,8 +3274,53 @@ def _struct_get_field(expr, field_name):
     return ops.StructField(expr, field_name).to_expr().name(field_name)
 
 
-def _destructure(expr: StructColumn) -> DestructColumn:
-    """ Destructure a ``Struct`` to create a destruct column.
+def _struct_get_attr(expr: StructValue, attr_name: str) -> ValueExpr:
+    """Get field named `attr_name` from the ``StructValue`` expression `expr`.
```

Comment:
is there a reason you are now just defining ```__getstate___``` and ```__setstate___``` on a StructValueExpr itself?

Reply:
this is *extremely* convoluted

Reply:
> is there a reason you are now just defining `__getstate___` and `__setstate___` on a StructValueExpr itself?

Do you mean defining `__getstate__` and `__setstate__` but mimic the default implementation (i.e. their behavior if they weren't defined), so that **`__getattr__`** is not triggered?

I think that would work too, but I'm not sure how much better it would be:
* It feels odd to be explicitly defining `__getstate__` and `__setstate__` with an implementation that is just their default implementation
* On the other hand, my current implementation is essentially defining **`__getattr__`** with an exclusion (that says that it should not be used for `__getstate__` and `__setstate__`), which seems a little easier to follow to me

I would agree that both solutions feel like hacks though